forc-plugins/forc-client/src/ops/tx_util.rs

```diff
@@ -155,6 +155,11 @@ impl TransactionBuilderExt for TransactionBuilder {
 
         if !unsigned {
             let signature = if let Some(signing_key) = signing_key {
+                // Safety: `Message::from_bytes_unchecked` is unsafe because
+                // it can't guarantee that the provided bytes will be the product
+                // of a cryptographically secure hash. However, the bytes are
+                // coming from `tx.id()`, which already uses `Hasher::hash()`
+                // to hash it using a secure hash mechanism.
                 let message = unsafe { Message::from_bytes_unchecked(*tx.id()) };
```

Comment:
Hmm, sounds good to me but I wonder if we should continue with this PR and track it with separate issues in rs-SDK, forc-wallet and sway repo until we got the change and release to fuel-tx, mentioned at your comment 🤔 Because as of now, without this change we cannot deploy anything signed which is pretty critical to fix.

cc @mitchmindtree @sezna

Reply:
Created #3224 for tracking this here, I will be opening up issues at forc-wallet and sdk too 👍